src/main/java/uk/gov/companieshouse/authcodenotification/utils/Encrypter.java

```diff
@@ -2,61 +2,55 @@
 import javax.crypto.Cipher;
-import javax.crypto.IllegalBlockSizeException;
-import javax.crypto.NoSuchPaddingException;
 import javax.crypto.spec.IvParameterSpec;
 import javax.crypto.spec.SecretKeySpec;
 
 import java.nio.charset.StandardCharsets;
-import java.security.InvalidAlgorithmParameterException;
-import java.security.InvalidKeyException;
-import java.security.NoSuchAlgorithmException;
 import java.security.SecureRandom;
 import java.util.Base64;
 
 import java.io.ByteArrayOutputStream;
 import java.io.IOException;
+import java.util.Map;
 
 @Component
 public class Encrypter {
 
     private static final String AES_CBC_PKCS_5_PADDING = "AES/CBC/PKCS5Padding";
     private static final String AES = "AES";
 
-    @Value("${AUTH_CODE_ENCRYPT_KEY}")
+    @Value("${AUTH_CODE_ENCRYPTION_KEY}")
     private String aesKeyString;
 
-    public String encrypt(String plainText) throws
-            NoSuchPaddingException,
-            NoSuchAlgorithmException,
-            InvalidAlgorithmParameterException,
-            InvalidKeyException,
-            IllegalBlockSizeException,
-            BadPaddingException,
-            IOException {
-        byte[] key = aesKeyString.getBytes(StandardCharsets.UTF_8);
-        byte[] plaintextBytes = plainText.getBytes(StandardCharsets.UTF_8);
+    public String encrypt(String requestId, String plainTextToEncrypt, Map<String, Object> logMap) throws ServiceException {
+        try {
+            byte[] key = aesKeyString.getBytes(StandardCharsets.UTF_8);
+            byte[] plainTextBytes = plainTextToEncrypt.getBytes(StandardCharsets.UTF_8);
 
-        var cipher = Cipher.getInstance(AES_CBC_PKCS_5_PADDING);
+            var cipher = Cipher.getInstance(AES_CBC_PKCS_5_PADDING);
 
-        var random = SecureRandom.getInstanceStrong();
-        var initialisationVector = new byte[Cipher.getInstance(AES_CBC_PKCS_5_PADDING).getBlockSize()];
-        random.nextBytes(initialisationVector);
+            var random = SecureRandom.getInstanceStrong();
+            var initialisationVector = new byte[Cipher.getInstance(AES_CBC_PKCS_5_PADDING).getBlockSize()];
+            random.nextBytes(initialisationVector);
 
-        cipher.init(Cipher.ENCRYPT_MODE, new SecretKeySpec(key, AES), new IvParameterSpec(initialisationVector));
-        byte[] encodedBytes = cipher.doFinal(plaintextBytes);
+            cipher.init(Cipher.ENCRYPT_MODE, new SecretKeySpec(key, AES), new IvParameterSpec(initialisationVector));
+            byte[] encodedBytes = cipher.doFinal(plainTextBytes);
 
-        byte[] encodedSaltedBytes = concatByte(initialisationVector, encodedBytes);
-        return Base64.getEncoder().encodeToString(encodedSaltedBytes);
+            byte[] encodedSaltedBytes = concatByte(initialisationVector, encodedBytes);
+            return Base64.getEncoder().encodeToString(encodedSaltedBytes);
+        } catch (Exception e) {
+            ApiLogger.errorContext(requestId, e.getMessage(), e, logMap);
```

Comment:
It's minor, but I would set a general, plain English message as the second parameter here, rather than use `e.getMessage()`. That will be output anyway as part of the stack trace.

"Error when encrypting text", for example.